### PR TITLE
feature/TRACEFOSS-2512 Frontend/Backend Global Search

### PR DESCRIPTION
**Pull request by ds-mwesener**

### CODE REVIEW

**Review by ds-ext-sceronik — APPROVED** [review]
LGTM :)

File: frontend/src/app/modules/page/parts/presentation/parts.component.ts [hunk 1]
```diff
@@ -79,20 +83,56 @@ export class PartsComponent implements OnInit, OnDestroy, AfterViewInit {
...
+    public searchControl: FormControl;
 
     public ngOnInit(): void {
         this.partsFacade.setPartsAsBuilt();
         this.partsFacade.setPartsAsPlanned();
+        this.searchFormGroup.addControl("partSearch", new FormControl([]));
+        this.searchControl = this.searchFormGroup.get('partSearch') as unknown as FormControl;
+        this.searchFormGroup.valueChanges.subscribe((formValues) => {
+            console.log("Change", formValues);
+        });
```
Comment (ds-mmaul) [inline frontend/src/app/modules/page/parts/presentation/parts.component.ts:96]: delete this

File: frontend/src/app/modules/page/parts/presentation/parts.component.ts [hunk 2]
```diff
@@ -79,20 +83,56 @@ export class PartsComponent implements OnInit, OnDestroy, AfterViewInit {
...
+            this.partsFacade.setPartsAsBuilt();
+            this.partsFacade.setPartsAsPlanned();
+        }
+
+    }
+
+
+    private resetLocalFilters(filterIsSet: boolean) {
+        for (const partTableComponent of this.partsTableComponents) {
+            if (partTableComponent.multiCompo.theSearchElement) {
```
Comment (ds-mmaul) [inline frontend/src/app/modules/page/parts/presentation/parts.component.ts:130]: maybe consider rename of multicompo

File: frontend/src/app/modules/shared/components/multi-select-autocomplete/multi-select-autocomplete.component.ts [hunk 3]
```diff
@@ -191,7 +201,10 @@ export class MultiSelectAutocompleteComponent implements OnChanges {
             this.selectAllChecked = selectedCount === this.filteredOptions.length;
         }
         this.selectedValue = val.value;
+        this.formControl.patchValue(val.value);
         this.selectionChange.emit(this.selectedValue);
+        this.theSearchElement = val.value;
+        console.log("selection change");
```
Comment (ds-mmaul) [inline frontend/src/app/modules/shared/components/multi-select-autocomplete/multi-select-autocomplete.component.ts:207]: delete

**Review by ds-mmaul — CHANGES_REQUESTED** [review]

**Comment by ds-mmaul** [comment]
> [**Trace-X Root**] Kudos, SonarCloud Quality Gate passed!    [Quality Gate passed](https://sonarcloud.io/dashboard?id=catenax-ng_tx-traceability-foss_backend&pullRequest=636)
> 
> [Bug](https://sonarcloud.io/project/issues?id=catenax-ng_tx-traceability-foss_backend&pullRequest=636&resolved=false&types=BUG) [A](https://sonarcloud.io/project/issues?id=catenax-ng_tx-traceability-foss_backend&pullRequest=636&resolved=false&types=BUG) [0 Bugs](https://sonarcloud.io/project/issues?id=catenax-ng_tx-traceability-foss_backend&pullRequest=636&resolved=false&types=BUG) [Vulnerability](https://sonarcloud.io/project/issues?id=catenax-ng_tx-traceability-foss_backend&pullRequest=636&resolved=false&types=VULNERABILITY) [A](https://sonarcloud.io/project/issues?id=catenax-ng_tx-traceability-foss_backend&pullRequest=636&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](https://sonarcloud.io/project/issues?id=catenax-ng_tx-traceability-foss_backend&pullRequest=636&resolved=false&types=VULNERABILITY) [Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=catenax-ng_tx-traceability-foss_backend&pullRequest=636&resolved=false&types=SECURITY_HOTSPOT) [A](https://sonarcloud.io/project/security_hotspots?id=catenax-ng_tx-traceability-foss_backend&pullRequest=636&resolved=false&types=SECURITY_HOTSPOT) [0 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=catenax-ng_tx-traceability-foss_backend&pullRequest=636&resolved=false&types=SECURITY_HOTSPOT) [Code Smell](https://sonarcloud.io/project/issues?id=catenax-ng_tx-traceability-foss_backend&pullRequest=636&resolved=false&types=CODE_SMELL) [A](https://sonarcloud.io/project/issues?id=catenax-ng_tx-traceability-foss_backend&pullRequest=636&resolved=false&types=CODE_SMELL) [5 Code Smells](https://sonarcloud.io/project/issues?id=catenax-ng_tx-traceability-foss_backend&pullRequest=636&resolved=false&types=CODE_SMELL)
> 
> [](https://sonarcloud.io/component_measures?id=catenax-ng_tx-traceability-foss_backend&pullRequest=636&metric=new_coverage&view=list) [88.7% Coverage](https://sonarcloud.io/component_measures?id=catenax-ng_tx-traceability-foss_backend&pullRequest=636&metric=new_coverage&view=list) [](https://sonarcloud.io/component_measures?id=catenax-ng_tx-traceability-foss_backend&pullRequest=636&metric=new_duplicated_lines_density&view=list) [0.0% Duplication](https://sonarcloud.io/component_measures?id=catenax-ng_tx-traceability-foss_backend&pullRequest=636&metric=new_duplicated_lines_density&view=list)

check code smells

**Review by ds-mmaul — APPROVED** [review]
LGTM

**Pull request merged**